docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+        As an Administrator
+        I want to add a new country to the store
+
+        Background:
+            Given I am logged in as an administrator
+
+        @ui
+        Scenario: Adding country
+            Given I want to add a new country
+            When I choose "France"
+            And I add it
+            Then I should be notified that it has been successfully created
+            And the country "France" should appear in the store
+
+Pay special attention to form of those sentences, they are specially constructed that way to hide details like ``I click on select box and i click on France country, I see France country in box with css .table``.
+The side effect of this is that we got more generic steps and when we want to change way of testing this feature for example in domain context it will be extremely easy all we need to do is adding ``@domain`` tag and of course implement proper steps in domain context.
+
+Choosing correct suite
+----------------------
+Next thing is creating suite, in our example at the beginning we decide that it will be user interface feature.
+So this suite will be placed in ``etc/behat/suites/ui/addressing/managing_countries.yml`` and it will looks like
+
+.. code-block:: yaml
+
+    default:
+        suites:
+            ui_managing_countries:
+                contexts_as_services:
+                    - sylius.behat.context.hook.doctrine_orm //This service is responsible for clearing database before scenario.
+
+                    - sylius.behat.context.transform.country
+                    - sylius.behat.context.transform.shared_storage
+                    // Those contexts services are responsible for all transformation for example  "And the country "France" should appear in the store" transforms "(the country "France")" to proper Country object.
+
+                    - sylius.behat.context.setup.geographical
+                    - sylius.behat.context.setup.security
+                    // Setup contexts in our case they are building our background scene. Preparing all actors.
+
+                    // Lights, Camera, Action!
+                    - sylius.behat.context.ui.admin.managing_countries
+                    - sylius.behat.context.ui.admin.notification
+                    // Those contexts are essential here we are placing all action steps like "When I choose "France" and I add it Then I should ne notified that...".
+                filters:
+                    tags: "@managing_countries && @ui"
+
+First essential thing that happens here is configuration of tags, from now behat will be searching all your features tagged with ``@managing_countires`` and your scenarios tagged with ``@ui``.
+Second thing is ``contexts_as_services:`` in this section we will be placing all our services with step implementation.
+
+I mentioned about side effects of ubiquitous language that's why we can very easy switch our testing context to ``@domain``. Domain suite will look like:
```

Comment:
`We have mentioned the with the generic steps we can easily switch our testing context to @domain. Have a look how it looks:`